app/controllers/drivers_controller.rb

```diff
@@ -0,0 +1,73 @@
+class DriversController < ApplicationController
+
+  def index
+    @drivers = Driver.order(:id)
+  end
+
+  def show
+    @driver = Driver.find_by(id: params[:id])
+
+    if @driver.nil?
+      redirect_to drivers_path
+      return
+    end
+  end
+
+  def new
+    @driver = Driver.new
+  end
+
+  def create
+    @driver = Driver.new(driver_params)
+
+    if @driver.save
+      redirect_to driver_path(@driver.id)
+      return
+    else
+      render :new
```

Comment:
It's best to include an HTTP status code. 

```suggestion
      render :new, status: :bad_request
```